Products/ZenEvents/zentrap.py

```diff
@@ -762,71 +766,79 @@ def decode(value):
-            else:
-                tz_mins = time.timezone / 60
-                if tz_mins < 0:
-                    utc_dir = '-'
-                    tz_mins = -tz_mins
+            Tries converting a DateAndTime value to a printable string.
+
+            A date-time specification.
+            field  octets  contents                  range
+            -----  ------  --------                  -----
+            1      1-2     year*                     0..65536
+            2        3     month                     1..12
+            3        4     day                       1..31
+            4        5     hour                      0..23
+            5        6     minutes                   0..59
+            6        7     seconds                   0..60
+                          (use 60 for leap-second)
+            7        8     deci-seconds              0..9
+            8        9     direction from UTC        '+' / '-'
+            9       10     hours from UTC*           0..13
+            10      11     minutes from UTC          0..59
+            """
+        try:
+            # Some traps send invalid UTC times (direction/hours/minutes all zeros)
+            if value[8:] == '\x00\x00\x00':
+                value = value[:8]
+            vallen = len(value)
+            if vallen == 8 or (vallen == 11 and value[8] in ('+', '-')):
+                (year, mon, day,
+                 hour, mins, secs, dsecs) = unpack(">HBBBBBB", value[:8])
+                # Ensure valid date representation
+                if mon < 1 or mon > 12:
+                    return None
+                if day < 1 or day > 31:
+                    return None
+                if hour < 0 or hour > 23:
+                    return None
+                if mins > 60:
+                    return None
+                if secs > 60:
+                    return None
+                if dsecs > 9:
+                    return None
+                if vallen == 11:
+                    utc_dir = value[8]
+                    (utc_hours, utc_mins) = unpack(">BB", value[9:])
                 else:
-                    utc_dir = '+'
-                utc_hours = tz_mins / 60
-                utc_mins = tz_mins % 60
-            return "%04d-%02d-%02dT%02d:%02d:%02d.%d00%s%02d:%02d" % (
-                year, mon, day, hour, mins, secs,
-                dsecs, utc_dir, utc_hours, utc_mins)
+                    tz_mins = time.timezone / 60
+                    if tz_mins < 0:
+                        utc_dir = '-'
+                        tz_mins = -tz_mins
+                    else:
+                        utc_dir = '+'
+                    utc_hours = tz_mins / 60
+                    utc_mins = tz_mins % 60
+                return "%04d-%02d-%02dT%02d:%02d:%02d.%d00%s%02d:%02d" % (
+                    year, mon, day, hour, mins, secs,
+                    dsecs, utc_dir, utc_hours, utc_mins)
+
+        except TypeError:
+            pass
```

Comment:
Perhaps these should be logged?  With this, I foresee future "where's the trap?" customer calls and we have nothing in our logs to show for it.